modules/SofaGeneralEngine/src/SofaGeneralEngine/NearestPointROI.inl

```diff
@@ -51,22 +53,55 @@ NearestPointROI<DataTypes>::~NearestPointROI()
 template <class DataTypes>
 void NearestPointROI<DataTypes>::init()
 {
-    if(!mstate1 || !mstate2) {
-        msg_error() << "Cannot Initialize without valid objects";
-        //mstate1->set(static_cast<simulation::Node*>(this->getContext())->getMechanicalState());
+    // Test inputs
+    bool success = true;
+    if (mstate1 && !mstate1.get())
+    {
+        msg_error_when(!mstate1.get()) << "Cannot Initialize, mstate1 link is pointing to invalid object!";
+        success = false;
+    }
+    else if (!mstate1)
+    {
+        msg_error_when(!mstate1) << "Cannot Initialize, mstate1 link is invalid!";
+        success = false;
+    }
+
+    if (mstate2 && !mstate2.get())
+    {
+        msg_error_when(!mstate2.get()) << "Cannot Initialize, mstate2 link is pointing to invalid object!";
+        success = false;
+    }
+    else if (!mstate2)
+    {
+        msg_error_when(!mstate2) << "Cannot Initialize, mstate2 link is invalid!";
+        success = false;
+    }
+
+    if(!success) {
```

Comment:
Cosmetic: put curly brace on the next line and add space between `if` and the parenthesis